src/cryptography/hazmat/primitives/ciphers/algorithms.py

```diff
@@ -24,7 +24,8 @@ def _verify_key_size(algorithm, key):
 class AES(object):
     name = "AES"
     block_size = 128
-    key_sizes = frozenset([128, 192, 256])
+    # 512 added to support AES-256-XTS, which uses 512-bit keys
+    key_sizes = frozenset([128, 192, 256, 512])
```

Comment:
You'd also need 384 for AES-192-XTS, right?

Reply:
OpenSSL doesn't support AES-192-XTS (and NIST also doesn't publish test vectors), so we won't be supporting that key size.

I should probably add a bit to the XTS mode docs about only being 128/256 (with 256/512 keys)

Reply:
I'm not wild about this. Once upon a time we thought about putting the key bytes in `XTS`. What factors led you to move away from this?

Do other crypto libraries do this the same way?

Reply:
Golang has a separate constructor for each mode, Crypto++ doesn't support XTS (or doesn't document it at least), and Botan has you get a cipher mode and then call `set_key` which needs to be the double length key.

I decided against having the "extra" key bytes be in the XTS mode constructor itself because it seems really bizarre to ask users to split the key in half and pass half to the AES object and half to the mode when it's all the key.